### ISSUE TITLE
Move type validation in core

### ISSUE THREAD

**Issue by bartv** [inmanta/inmanta-core#6540]
Pydantic v2 dropped support for constr of with a regex constraint. This has been replaced by pattern. However, pattern is a linear time rust implementation of regexes. It does not support the same regexes as the python flavor we have been using.

The goal is that both in std and inmanta-lsm we rewrite constr(regex=...) to using a special regex alternative. This alternative is explained in https://docs.pydantic.dev/latest/migration/#patterns-regex-on-strings

The goal of this ticket is to:

- [x] Add a method to core that can be used by both std and inmanta-lsm to validate a type, i.e. move most of `inmanta_plugins.std.validate_type` into this new method. Implement this method in iso6 and iso7 both.
- [x] update std to use this new method.

Once this is done, inmanta-lsm can be updated to use this (not part of this ticket). This needs to be in place, before we can merge pydantic v2 support.

In solutions the following MR also depend on it: https://code.inmanta.com/solutions/modules/config/-/merge_requests/5
